adam-core/src/main/scala/org/bdgenomics/adam/models/ReadBucket.scala

```diff
@@ -96,9 +96,9 @@ class ReadBucketSerializer extends Serializer[ReadBucket] {
 object ReadBucket {
   implicit def singleReadBucketToReadBucket(bucket: SingleReadBucket): ReadBucket = {
     val (pairedPrimary, unpairedPrimary) = bucket.primaryMapped.partition(_.getReadPaired)
-    val (pairedFirstPrimary, pairedSecondPrimary) = pairedPrimary.partition(_.getFirstOfPair)
+    val (pairedFirstPrimary, pairedSecondPrimary) = pairedPrimary.partition(_.getReadNum == 0)
```

Comment:
Now that `getReadNum` is more generic. We should throw an error here, if `getReadNum > 2`, correct?